.github/workflows/ci.yaml

```diff
@@ -21,10 +19,18 @@ jobs:
       - name: Build
         run: go build -v ./...
 
-      - name: Test
+      - name: Run Test
         env:
           MQTT_BROKER_URL: "tcp://localhost:1883"
-        run: go test -race -v ./...
+        run: go test -race -v -covermode atomic -coverprofile=covprofile ./...
+
+      - name: Install goveralls
```

Comment:
Apparently there's a github action available as well that has some nice features, like commenting on the PR with the change in coverage. Might be worth exploring if it's easy enough to add here

Reply:
https://github.com/coverallsapp/github-action

Reply:
I'll do that separately